processing/src/main/java/org/apache/druid/segment/incremental/ParseExceptionReport.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.segment.incremental;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.druid.query.ordering.StringComparators;
+
+import java.util.List;
+import java.util.Objects;
+
+public class ParseExceptionReport implements Comparable<ParseExceptionReport>
+{
+  private final String input;
+  private final String errorType;
+  private final List<String> details;
+  private final long timeOfExceptionMillis;
+
+  @JsonCreator
+  public ParseExceptionReport(
+      @JsonProperty("input") String input,
+      @JsonProperty("errorType") String errorType,
+      @JsonProperty("details") List<String> details,
+      @JsonProperty("timeOfExceptionMillis") long timeOfExceptionMillis
+  )
+  {
+    this.input = input;
+    this.errorType = errorType;
+    this.details = details;
+    this.timeOfExceptionMillis = timeOfExceptionMillis;
+  }
+
+  @JsonProperty
+  public String getInput()
+  {
+    return input;
+  }
+
+  @JsonProperty
+  public String getErrorType()
+  {
+    return errorType;
+  }
+
+  @JsonProperty
+  public List<String> getDetails()
+  {
+    return details;
+  }
+
+  @JsonProperty
+  public long getTimeOfExceptionMillis()
+  {
+    return timeOfExceptionMillis;
+  }
+
+  @Override
+  public int compareTo(ParseExceptionReport o)
```

Comment:
```
     * <p>It is strongly recommended, but <i>not</i> strictly required that
     * <tt>(x.compareTo(y)==0) == (x.equals(y))</tt>.  Generally speaking, any
     * class that implements the <tt>Comparable</tt> interface and violates
     * this condition should clearly indicate this fact.  The recommended
     * language is "Note: this class has a natural ordering that is
     * inconsistent with equals."
    ```
From `compareTo` documentation. do we not want to include `details` when doing the comparison? If not, maybe it is better to create a separate comparator class and pass it to `TreeSet` than overriding `compareTo` here.

Reply:
There shouldn't be any parse exceptions with identical input/type/timestamp but different details, so I left that out of the comparison, I moved this to a comparator